src/OracleService/Protocols/OracleHttpsProtocol.cs

```diff
@@ -40,15 +40,28 @@ public async Task<(OracleResponseCode, string)> ProcessAsync(Uri uri, Cancellati
 
             if (!Settings.Default.AllowPrivateHost)
             {
-                IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host);
+                IPHostEntry entry = await Dns.GetHostEntryAsync(uri.Host, cancellation);
                 if (entry.IsInternal())
                     return (OracleResponseCode.Forbidden, null);
             }
 
             HttpResponseMessage message;
             try
             {
+            download:
                 message = await client.GetAsync(uri, HttpCompletionOption.ResponseContentRead, cancellation);
+                if (message.StatusCode == HttpStatusCode.NotFound && message.Headers.Location is not null)
```

Comment:
should it be `Found`?

Reply:
Maybe is not required to check the `StatusCode`